components/steps/Steps.tsx

```diff
@@ -1,17 +1,23 @@
-import Step1 from '@/components/steps/Step1'
-import Step2 from '@/components/steps/Step2'
-import Step3 from '@/components/steps/Step3'
-import Step4 from '@/components/steps/Step4'
+import { Step1Icon, Step2Icon, Step3Icon, Step4Icon, useStepIcon } from '@/components/steps/icons'
 
-export default function Steps() {
+type StepsProps = {
+  activeStep: number
+}
+
+export function Steps({ activeStep }: StepsProps) {
   return (
     <section>
       <div className='flex flex-row gap-x-4 justify-end'>
-        <Step1 height={48} width={48} fill='#671bc9' fillRule='nonzero' stroke='black' />
-        <Step2 height={48} width={48} />
-        <Step3 height={48} width={48} />
-        <Step4 height={48} width={48} />
+        {[Step1Icon, Step2Icon, Step3Icon, Step4Icon].map((StepIcon, idx) => (
+          <StepIcon className='w-12 h-12' {...useStepIcon(idx, activeStep)} />
+        ))}
```

Comment:
Thanks to this change we can easily get the index (idx) of a given StepIcon and use it for calculating the props for that icon.